.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -732,36 +732,119 @@ private string ParseTimePeriod(Match match, out int beginHour, out int endHour)
             }
 
             // Parse "pm"
-            var matchPmStr = match.Groups[Constants.PmGroupName].Value;
-            var matchAmStr = match.Groups[Constants.AmGroupName].Value;
-            var descStr = match.Groups[Constants.DescGroupName].Value;
-            if (!string.IsNullOrEmpty(matchAmStr) || (!string.IsNullOrEmpty(descStr) && descStr.StartsWith("a")))
+            string matchPmStr = string.Empty, matchAmStr = string.Empty, descStr = string.Empty;
+            string matchPmStr1 = string.Empty, matchAmStr1 = string.Empty, descStr1 = string.Empty;
+            var matchPmStrGroup = match.Groups[Constants.PmGroupName];
+            var matchAmStrGroup = match.Groups[Constants.AmGroupName];
+            var descStrGroup = match.Groups[Constants.DescGroupName];
+            if (matchPmStrGroup.Captures.Count > 0)
             {
-                if (beginHour >= Constants.HalfDayHourCount)
+                matchPmStr = matchPmStrGroup.Captures[0].Value;
```

Comment:
It is better to change the patterns to use named capture groups. Using indices can lead to issues as different patterns are edited.
Also, can you give better names to variables than matchPmStr1? What's the key difference from matchPmStr?
Lastly, why is this change not necessary in English? Doesn't English have corresponding test specs? IF not, please add one sentence to demonstrate this new code also works there, ok?